### PR TITLE
ParserTest

### LINKED ISSUE #374: test Parser class

### PR DESCRIPTION
**Pull request by kaboomzxc**
51 tests case passed, 100% class coverage, 100% method coverage, 70% line coverage, 92% branch coverage.

Closes #374

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -12,6 +12,9 @@ repositories {
 dependencies {
     testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.10.0'
     testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.10.0'
+
+    testImplementation 'org.mockito:mockito-core:5.3.1'
+    testImplementation 'org.mockito:mockito-junit-jupiter:5.3.1'
```
Comment (yentheng0110) [inline build.gradle:17]: I saw from this link: https://github.com/nus-cs2113-AY2425S1/forum/issues/36
Mockito library is approved for use in tP!

File: src/test/java/bookbob/parser/ParserTest.java [hunk 2]
```diff
@@ -0,0 +1,978 @@
...
+        String input = "exit";
+
+        // Act
+        boolean result = Parser.handleCommand(input, commandHandler, records, appointmentRecord);
+
+        // Assert
+        Mockito.verify(commandHandler, Mockito.times(1)).removePastAppointments(appointmentRecord);
+        Mockito.verify(commandHandler, Mockito.times(1)).exit(input);
+        Assertions.assertTrue(result);
+    }
```
Comment (yentheng0110) [inline src/test/java/bookbob/parser/ParserTest.java:224]: Perhaps can consider moving this class into a class called ExitCommandTest

File: src/test/java/bookbob/parser/ParserTest.java [hunk 3]
```diff
@@ -0,0 +1,978 @@
...
+            // Assert
+            Mockito.verify(commandHandler, Mockito.times(1)).deleteAppointment(input, appointmentRecord);
+            Assertions.assertTrue(result);
+        }
+    }
+
+    //@@author kaboomzxc
+    @Nested
+    @DisplayName("Search Command Tests")
+    class SearchCommandTests {
```
Comment (yentheng0110) [inline src/test/java/bookbob/parser/ParserTest.java:158]: Perhaps can rename this as FindVisitCommandTests

**Review by yentheng0110 — COMMENTED** [review]
Overall LGTM, left some suggestions for you to consider

File: src/test/java/bookbob/parser/ParserTest.java (hunk 3 above)
Comment (kaboomzxc) [reply src/test/java/bookbob/parser/ParserTest.java:158]: actually for this part, not really testing findVisit only, was also testing "findDiagnosis" and "findMedication" also

but i renamed to "find" which suits better

File: build.gradle (hunk 1 above)
Comment (kaboomzxc) [reply build.gradle:17]: yes that is great !

File: src/test/java/bookbob/parser/ParserTest.java (hunk 3 above)
Comment (yentheng0110) [reply src/test/java/bookbob/parser/ParserTest.java]: Yup sure

**Pull request merged**